wasm/src/main/scala/ir2wasm/HelperFunctions.scala

```diff
@@ -1655,8 +1657,40 @@ object HelperFunctions {
       instrs += I32_CONST(1)
       instrs += RETURN
     } // test fail
-    instrs += DROP
-    instrs += I32_CONST(0) // false
+
+    if (classInfo.isAncestorOfHijackedClass) {
+      /* It could be a hijacked class instance that implements this interface.
+       * Test whether `jsValueType(expr)` is in the `specialInstanceTypes` bitset.
+       * In other words, return `(1 << jsValueType(expr)) != 0`.
+       */
```

Comment:
Though it might be a bit duplicated with an explanation at `specialInstanceTypes`, it would be helpful for me if we can add some example like below here 😅 

```suggestion
       *
       * For example, when the `jsValueType(expr)` is 'string', 'expr.isInstanceOf[j.l.CharSequence]'
       * should be `true` because `j.l.String` implements `j.l.CharSequence`, and
       * `specialInstanceTypes` for `j.l.CharSequence` contains `1 << jsValueType(j.l.String)`.
       */
```

Reply:
I added a more concrete example. :)